src/asmjit/core/virtmem.cpp

```diff
@@ -300,7 +302,10 @@ class AnonymousMemory {
     static volatile uint32_t memfd_create_not_supported;
 
     if (!memfd_create_not_supported) {
-      _fd = (int)syscall(__NR_memfd_create, "vmem", 0);
+      int flags = 0;
+      if (Support::test(memoryFlags, MemoryFlags::kMMapEnableMapHugeTlb))
+          flags |= MFD_HUGETLB | MFD_HUGE_2MB;
```

Comment:
I guess we don't have to worry about not having these constants available right? I never tried to compile asmjit on a very old kernel anyway, seems this is OK.

Reply:
I am far from being a expert in Linux distribution archeology, but it seemed to be available on the Debian, CentOS, and Fedora machines I have access to.  It just occurred to me that I could replace the use of `syscall` with the `memfd_create` function call, for consistency.  If we're relying on the constants, we can rely on that function prototype, too.